### ISSUE TITLE
[New rule] Prefer variable instead of multiple function calls

### ISSUE THREAD

**Issue by NickBalnaves** [dart-code-checker/dart-code-metrics#746]
**Please describe what the rule should do:**
Prefer declaring a variable when calling a function mutliple times. 

**What category of rule is this? (place an "X" next to just one item)**
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**
Bad:
```dart
return Container(
  color: Theme.of(context).colorScheme.secondary,
  child: Text(
    'Text with a background color',
    style: Theme.of(context).textTheme.headline6,
  ),
);
```

Good:
```dart
final theme = Theme.of(context);
return Container(
  color: theme.colorScheme.secondary,
  child: Text(
    'Text with a background color',
    style: theme.textTheme.headline6,
  ),
);
```

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by incendial** [comment]
Really interesting idea, thanks for filling the issue!

**Comment by incendial** [comment]
Available in 4.14.0 🚀
